src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SMKParamsReference.kt

```diff
@@ -0,0 +1,54 @@
+package com.jetbrains.snakecharm.lang.psi
+
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiPolyVariantReference
+import com.intellij.psi.PsiReferenceBase
+import com.intellij.psi.ResolveResult
+import com.intellij.psi.util.parentOfType
+import com.jetbrains.python.psi.resolve.RatedResolveResult
+import java.util.*
+
+
+class SMKParamsReference(
+        element: PsiElement, textRange: TextRange
+) : PsiReferenceBase<PsiElement>(element, textRange), PsiPolyVariantReference {
+    private val key: String = element.text.substring(textRange.startOffset, textRange.endOffset)
+
+    override fun multiResolve(incompleteCode: Boolean): Array<ResolveResult> {
+        val paramsSection = getParamsSection() ?: return emptyArray()
+        paramsSection.argumentList?.arguments?.forEach {
```

Comment:
Code duplication here and in get variants method. Please extract method which collects keyword arguments.